robottelo/datafactory.py

```diff
@@ -0,0 +1,163 @@
+# -*- encoding: utf-8 -*-
+"""Data Factory for all entities"""
+import random
+
+from functools import wraps
+from fauxfactory import gen_string, gen_integer
+
+from robottelo.config import settings
+from robottelo.decorators import bz_bug_is_open
+
+
+class InvalidArgumentError(Exception):
+    """Indicates an error when an invalid argument is received."""
+
+
+def smoke(func):
+    """Overrides the data creator functions in this class to return 1 value
+
+    If smoke = True, return a random data.
+    If smoke = False, return the entire data set.
+
+    """
+    @wraps(func)
+    def func_wrapper(*args, **kwargs):
+        """Perform smoke test attribute check"""
+        dataset = func(*args, **kwargs)
+        if settings.smoke:
+            dataset = [random.choice(dataset)]
+        return dataset
+    return func_wrapper
+
+
+@smoke
+def generate_strings_list(len1=None, remove_str=None, bug_id=None):
```

Comment:
@JacobCallahan good suggestion.  But I may have to fix this in next PR (tracking here: https://github.com/SatelliteQE/robottelo/issues/2996) since this PR already has touched 71 files.  If I fix this, it may be more code to review.